<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>981</width>
    <height>835</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QFrame" name="frame_keys">
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_9">
         <item row="2" column="1">
          <widget class="QPushButton" name="pushButton_send">
           <property name="text">
            <string>послать</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0" rowspan="3">
          <widget class="QPlainTextEdit" name="plainTextEdit_log">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="centerOnScroll">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLineEdit" name="lineEdit_text"/>
         </item>
         <item row="3" column="1">
          <widget class="QPushButton" name="pushButton_clear">
           <property name="text">
            <string>очистить</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>471</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="groupBox_settings">
        <property name="title">
         <string>настройки:</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="1">
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>Управление колёсами:</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="2" column="2">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_right_wheel">
              <property name="decimals">
               <number>3</number>
              </property>
              <property name="minimum">
               <double>-1.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.001000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>правое:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSlider" name="horizontalSlider_right_wheel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSlider" name="horizontalSlider_left_wheel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksAbove</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_left_wheel">
              <property name="decimals">
               <number>3</number>
              </property>
              <property name="minimum">
               <double>-1.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.001000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>левое:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0" colspan="3">
             <layout class="QHBoxLayout" name="horizontalLayout">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_clear_power">
                <property name="text">
                 <string>Сброс</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="0" column="0" colspan="3">
             <widget class="QCheckBox" name="checkBox_enable_key_controll">
              <property name="text">
               <string>управление через клавиши: AWDS</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Коэффициенты PID:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QRadioButton" name="radioButton_angle">
              <property name="text">
               <string>Угол</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButton_left_wheel">
              <property name="text">
               <string>Скорость левого колеса</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButton_right_wheel">
              <property name="text">
               <string>Скорость правого колеса</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>P:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="horizontalSlider_p">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="pageStep">
                 <number>10</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="tickPosition">
                 <enum>QSlider::TicksAbove</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="doubleSpinBox_p">
                <property name="decimals">
                 <number>6</number>
                </property>
                <property name="minimum">
                 <double>-5.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>5.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>I:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="horizontalSlider_i">
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="pageStep">
                 <number>10</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="tickPosition">
                 <enum>QSlider::TicksAbove</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="doubleSpinBox_i">
                <property name="decimals">
                 <number>6</number>
                </property>
                <property name="minimum">
                 <double>-5.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>5.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>D:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="horizontalSlider_d">
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="pageStep">
                 <number>10</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="tickPosition">
                 <enum>QSlider::TicksAbove</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="doubleSpinBox_d">
                <property name="decimals">
                 <number>6</number>
                </property>
                <property name="minimum">
                 <double>-5.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>5.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>порт:</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="4">
             <widget class="QPushButton" name="pushButton_connect">
              <property name="text">
               <string>Connect</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinBox_port_name">
              <property name="value">
               <number>6</number>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="2">
             <widget class="QLineEdit" name="lineEdit_speed">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Speed&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="inputMask">
               <string>99999999; </string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>угол:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QDial" name="dial_angle">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;направление&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="maximum">
               <number>2000</number>
              </property>
              <property name="pageStep">
               <number>0</number>
              </property>
              <property name="invertedAppearance">
               <bool>false</bool>
              </property>
              <property name="invertedControls">
               <bool>false</bool>
              </property>
              <property name="wrapping">
               <bool>true</bool>
              </property>
              <property name="notchTarget">
               <double>3.700000000000000</double>
              </property>
              <property name="notchesVisible">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBox_angle">
              <property name="decimals">
               <number>4</number>
              </property>
              <property name="minimum">
               <double>-3.141500000000000</double>
              </property>
              <property name="maximum">
               <double>3.141500000000000</double>
              </property>
              <property name="singleStep">
               <double>0.010000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Смещение:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="horizontalSlider_offset">
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="tickPosition">
                 <enum>QSlider::TicksAbove</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="doubleSpinBox_offset">
                <property name="decimals">
                 <number>4</number>
                </property>
                <property name="minimum">
                 <double>-1.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_stop">
                <property name="text">
                 <string>Стоп</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item row="10" column="1" colspan="2">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="checkBox_enable_debug">
           <property name="text">
            <string>Разрешить отладку</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0" colspan="2">
          <widget class="QGroupBox" name="groupBox_9">
           <property name="title">
            <string>GroupBox</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QRadioButton" name="radioButton_left_wheel_speed">
              <property name="text">
               <string>Скорость левого колеса</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButton_right_wheel_speed">
              <property name="text">
               <string>Скорость правого колеса</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_15">
              <item>
               <widget class="QLabel" name="label_17">
                <property name="text">
                 <string>Скорость:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="horizontalSlider_speed">
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>233003000</number>
                </property>
                <property name="pageStep">
                 <number>10</number>
                </property>
                <property name="tracking">
                 <bool>true</bool>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="tickPosition">
                 <enum>QSlider::NoTicks</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox">
                <property name="maximum">
                 <number>233003000</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QFrame" name="frame">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>50</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QGridLayout" name="gridLayout_3"/>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>981</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAfqk">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="action_save_as"/>
    <addaction name="action_load"/>
    <addaction name="action_save"/>
   </widget>
   <addaction name="menuAfqk"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_save_as">
   <property name="text">
    <string>Сохранить как</string>
   </property>
  </action>
  <action name="action_load">
   <property name="text">
    <string>Открыть</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>Сохранить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_remove">
   <property name="text">
    <string>удлить</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_create">
   <property name="text">
    <string>создать</string>
   </property>
   <property name="shortcut">
    <string>Return</string>
   </property>
  </action>
  <action name="action_copy">
   <property name="text">
    <string>копировать</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="action_insert">
   <property name="text">
    <string>вставить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionAction_save_ranges">
   <property name="text">
    <string>action_save_ranges</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_clear</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEdit_log</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>475</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>419</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_speed</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>912</x>
     <y>714</y>
    </hint>
    <hint type="destinationlabel">
     <x>885</x>
     <y>714</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_speed</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>720</x>
     <y>719</y>
    </hint>
    <hint type="destinationlabel">
     <x>920</x>
     <y>712</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
